lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
the anonymous proc is private. and does not need to be tested. Maybe for this I propose an integration test. But I feel this is ok since the "integration" test is only testing the public interfaces of a single class.

`QueryRelation` tests ensure that calling these methods populate the hash.
`MyModel` test can ignore `QueryRelation` and just focuses that if you call it, it will call find correctly.

Yea, I think changing the `self` going in would be better. Possibly passing in `:klass`.

```ruby
describe QueryRelation do
  describe "#all" do
    it "yields" do
      model = double("Model", :klass => "Model")
      expect(model).to receive("rslt=").with(:select => %w(abc)).and_return([1])
      expect(QueryRelation.new(model).select("abc").to_a).to eq([1])
    end
  end
end
```